promtail: panic with external_labels

**Describe the bug**
Incorrect hyphen in yaml config leads to panic error.

**To Reproduce**
1. Add to client section the following section:  
  ```yaml
  clients:
    - url: https://need.host.com:3100/loki/api/v1/push
    - external_labels:
        host: host1
  ```
2. Start promtail.
3. See error in journalctl.

**Expected behavior**
Log contains clear error message.

**Systemd log**
```
Started Promtail for loki.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x124a17a]
goroutine 1 [running]:
github.com/grafana/loki/pkg/promtail/client.New(0x0, 0x3b9aca00, 0x19000, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /src/loki/pkg/promtail/client/client.go:107 +0x3a
github.com/grafana/loki/pkg/promtail/client.NewMulti(0x2dee8e0, 0xc00037b770, 0xc000184820, 0x2, 0x2, 0xc00011a140, 0x0, 0x0, 0xc0005
        /src/loki/pkg/promtail/client/multi.go:23 +0x140
github.com/grafana/loki/pkg/promtail.New(0x0, 0x0, 0x0, 0x0, 0x2378, 0x0, 0x0, 0x2387, 0x1, 0x6fc23ac00, ...)
        /src/loki/pkg/promtail/promtail.go:33 +0x1e1
main.main()
        /src/loki/cmd/promtail/main.go:59 +0x262
 Main process exited, code=exited, status=2/INVALIDARGUMENT
 Failed with result 'exit-code'.
```

--- a/pkg/promtail/client/client.go
+++ b/pkg/promtail/client/client.go
@@ -4,6 +4,7 @@ import (
 	"bufio"
 	"bytes"
 	"context"
+	"errors"
 	"fmt"
 	"io"
 	"net/http"
@@ -103,6 +104,10 @@ type entry struct {
 
 // New makes a new Client.
 func New(cfg Config, logger log.Logger) (Client, error) {
+	if cfg.URL.URL == nil {
+		return nil, errors.New("client needs target URL")
+	}
+
 	c := &client{
 		logger:  log.With(logger, "component", "client", "host", cfg.URL.Host),
 		cfg:     cfg,

